File: scripts/nodejs/06_example.mjs
// Real-world mini example: summarize text from stdin in one paragraph.
import OpenAI from "openai";

const text = await new Promise((resolve) => {
  let buf = "";
  process.stdin.setEncoding("utf8");
  process.stdin.on("data", (c) => (buf += c));
  process.stdin.on("end", () => resolve(buf.trim()));
});

if (!text) {
  console.error("usage: cat file.txt | node 06_example.mjs");
  process.exit(1);
}

const client = new OpenAI({ baseURL: "http://localhost:11434/v1", apiKey: "not-needed" });

const response = await client.chat.completions.create({
  model: "apple-foundationmodel",
  messages: [
    { role: "system", content: "You are a concise summarizer. Reply with one short paragraph." },
    { role: "user", content: `Summarize:\n\n${text}` },
  ],
  max_tokens: 150,
});

console.log((response.choices[0].message.content || "").trim());

File: scripts/python/06_example.py
#!/usr/bin/env python3
"""Real-world mini example: summarize text from stdin in one paragraph."""
import sys

from openai import OpenAI

text = sys.stdin.read().strip()
if not text:
    sys.exit("usage: cat file.txt | python 06_example.py")

client = OpenAI(base_url="http://localhost:11434/v1", api_key="not-needed")

response = client.chat.completions.create(
    model="apple-foundationmodel",
    messages=[
        {"role": "system", "content": "You are a concise summarizer. Reply with one short paragraph."},
        {"role": "user", "content": f"Summarize:\n\n{text}"},
    ],
    max_tokens=150,
)

print((response.choices[0].message.content or "").strip())
